### ISSUE TITLE
Unable to cast from Raw to long: 'QUEUED' when delete document with 'MULTI'

### ISSUE THREAD

**Issue by hizliemre** [StackExchange/StackExchange.Redis#1569]
When i want delete my documents in MULTI - EXEC blocks is raised this error : Unable to cast from Raw to long: 'QUEUED'

**Comment by hizliemre** [comment]
I fixed my issue with run 'FT.DEL' command manually.

**Comment by mgravell** [comment]
Hi; I'm glad you got a workaround, but is there a minimal repro for what
you had originally? That sounds very odd... I'd be very curious about
seeing the original code.

**Comment by hizliemre** [comment]
My code's summary is like this;

`db.Execute("MULTI");`
`client.DeleteDocument(doc);` 
`db.Execute("EXEC");`

**Comment by mgravell** [comment]
K, basically: don't do that. At the moment, SE-Redis is based on a multiplexer implementation, meaning that it is designed such that concurrent callers share the same physical connection. If you start issuing MULTI/EXEC, you have extremely undefined behavior.

There is a managed transaction API for MULTI/EXEC usage - not at a PC, but I want to say "create transaction" or "begin transaction". Use those instead of issuing your own MULTI

**Comment by hizliemre** [comment]
Thank you for your advice. But i couldn't use se-redis transaction api with nredisearch. Can u show an example for me?

**Comment by mgravell** [comment]
Indeed, NRediSearch is not designed with transactions in mind. I'd love to
discuss more of what you're actually trying to do here, so that we can see
whether this is a problematic gap or not.

Forgetting about implementation details: what is it that you're trying to
do atomically here?

**Comment by mgravell** [comment]
Thinking aloud: most simple operations should theoretically work fine *if*
they use the async API (which is up to the caller) *and* we can inject the
transaction context into the client (which should just be a constructor
thing). So it isn't insurmountable - I just want to make sure I understand
the use case.

**Comment by hizliemre** [comment]
I am trying write an orm similar entityframework over nredisearch. so i tried implement begintransaction and committransaction. You can check my context codes here : [https://github.com/hizliemre/redisworkcore/blob/master/RedisworkCore/RedisContext.cs](redisworkcore).

I have to remove this feature as i understand it from your advices. Is it right?

**Comment by mgravell** [comment]
I'm not saying it can't be done. I'm saying we might need some minor
construction changes. Let me review what you have and I'll get back to you.
